File: 251003/Novichenko/RV1/src/main/java/com/example/rv1/dto/UserDTO.java
package com.example.rv1.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import javax.validation.constraints.Size;

@Data
public class UserDTO {
    private int id;

    @JsonProperty("login")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String login;

    @JsonProperty("password")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String password;

    @JsonProperty("firstname")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String firstName;

    @JsonProperty("lastname")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String lastName;
}

File: 251002/Belyatskiy/2222/lab1/src/dtos/user.dto.ts
import { IsString, IsNotEmpty, Length } from 'class-validator';

export class UserRequestTo {
    @IsString()
    @IsNotEmpty()
    @Length(2, 64)
    login: string;

    @IsString()
    @IsNotEmpty()
    @Length(8, 128)
    password: string;

    @IsString()
    @IsNotEmpty()
    @Length(2, 64)
    firstname: string;

    @IsString()
    @IsNotEmpty()
    @Length(2, 64)
    lastname: string;
}

export class UserResponseTo {
    id: number;
    @Length(2, 64)
    login: string;
    firstname: string;
    lastname: string;
}